_posts/2021-11-18-real-time-and-historical-ad.markdown

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "Anomaly Detection for Historical Data and Real-time Streaming"
+authors:
+  - ohltyler
+  - bpavani
+date: 2021-11-18 09:00:00 -0700
+categories:
+  - technical-post
+twittercard:
+  description: "An overview on configuring and creating detectors for real-time and historical anomaly detection in OpenSearch."
+---
+
+You can leverage anomaly detection to analyze vast amount of logs in many different ways. Some analytics approaches require real-time detection, such as application monitoring, event detection, and fraud detection. Others involve analyzing past data to identify the trends and patterns, isolate the root cause and prevent them from happening again in the future. The anomaly detection plugin in OpenSearch already supports real-time streaming data. In OpenSearch 1.1, we introduced support for anomaly detection on historical data. Often-times, the historical and real-time use cases work hand-in-hand. A user may find real-time anomalies in their data, and decide to run historical detection to search for similar patterns in the past. Or, a user may first search for anomalies in the past data, then decide to configure anomaly detection to monitor their system for anomalies in real-time. OpenSearch 1.1 also streamlines the anomaly detection configuration with the new unified flow that allows users to configure the anomaly detector once that can then be applied to both real-time or historical analysis. In this blog, we will review each of the steps in the unified workflow with an example.
+
+Consider index `service_1_metrics` which contains different performance metrics for a service named `service_1`. The index `service_1_metrics` contains the fields `host`, `cpu_utilization`, `mem_utilization`, and `timestamp`, where `host` is the specific host that’s reporting the metrics, `cpu_utilization` is the current CPU utilization percentage reported by that host, `mem_utilization` is the current memory utilization percentage, and `timestamp` is the time when the metrics were collected. Now, an operator is interested in setting up anomaly detection on the CPU utilization and memory utilization metrics.
```

Comment:
Can we personalize the last sentence to be about the reader instead of an unnamed 'operator'?